zilliqa/src/contracts/deposit.sol

```diff
@@ -1,26 +1,138 @@
 // SPDX-License-Identifier: MIT OR Apache-2.0
 pragma solidity ^0.8.20;
 
+// Implementation of a circular queue/buffer of validator public keys
```

Comment:
I wonder if we could use simply dynamic arrays of type `Value[]` instead, with elements being pushed during the epoch and popped until the array is empty when `tickEpoch()` is called.